pkg/plugin/agent.go

```diff
@@ -0,0 +1,348 @@
+		io.Copy(os.Stdout, reader)
+	}
+	if err != nil {
+		log.Printf("[Plugin agent] Unable to pull image from container registry, got following error: %v\n", err)
+	}
+
+	resp, err = containerCreate(ctx, cli, cfg)
+	if err != nil {
+		return resp, ctx, cli, err
+	}
+
+	return resp, ctx, cli, err
+}
+
+func containerCreate(ctx context.Context, cli *client.Client, cfg *Config) (container.ContainerCreateCreatedBody, error) {
+	return cli.ContainerCreate(ctx,
+		&container.Config{
+			Image: cfg.DockerImg,
+			Cmd:   []string{GetShell(cfg)},
+			Tty:   true,
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeBind,
+					Source: cfg.InDir,
+					Target: "/input",
+				},
+			},
+			NetworkMode: "host",
+		}, nil, "")
+}
+
+// GetRegistryAuth returns authentication string required to pull container from container registry
+func GetRegistryAuth() (string, error) {
+	user := os.Getenv(envDockerUser)
+	token := os.Getenv(envDockerToken)
+
+	if user == "" || token == "" {
+		log.Println("[Plugin agent] No authentication credentials provided, please check if environment variables are set")
+		return "", errors.New("no authentication credentials provided")
+	}
+
+	authConfig := types.AuthConfig{
+		Username: user,
+		Password: token,
+	}
+
+	encodedJSON, err := json.Marshal(authConfig)
+	if err != nil {
+		return "", err
+	}
+
+	authStr := base64.URLEncoding.EncodeToString(encodedJSON)
+	return authStr, nil
+}
+
+// GetShell returns the Unix shell required to run the command lines
+func GetShell(cfg *Config) string {
+	return strings.Split(cfg.Cmd, " ")[0]
+}
+
+// ExecAllPluginCmd executes all required command lines in the container
+func ExecAllPluginCmd(ctx context.Context, containerID string, cfg *Config) error {
+	logFile, err := CreateLogFile()
+	if err != nil {
+		return err
+	}
+
+	currentCmd := "echo test"
+	expectedOutput := "test\n"
+	err = ExecPluginCmd(ctx, containerID, cfg, currentCmd, expectedOutput, true, logFile)
+	if err != nil {
+		return err
+	}
+
+	currentCmd = "mkdir /result"
+	expectedOutput = ""
+	err = ExecPluginCmd(ctx, containerID, cfg, currentCmd, expectedOutput, false, logFile)
+	if err != nil {
+		return err
+	}
+
+	currentCmd = cfg.Cmd[strings.Index(cfg.Cmd, "-c")+3 : len(cfg.Cmd)]
+	expectedOutput = ""
+	err = ExecPluginCmd(ctx, containerID, cfg, currentCmd, expectedOutput, false, logFile)
+	if err != nil {
+		return err
+	}
+
+	currentCmd = fmt.Sprintf("for i in /result/*; do curl -F name=%s -F result=@$i http://127.0.0.1:%d/save; done", cfg.Name, GetPortNumber())
+	expectedOutput = ""
+	err = ExecPluginCmd(ctx, containerID, cfg, currentCmd, expectedOutput, false, logFile)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[Plugin agent] All commands were executed, check logfile %v for outputs of executed command lines\n", logFile)
+
+	return nil
+}
+
+func CreateLogFile() (string, error) {
+	file, err := ioutil.TempFile("", time.Now().Format("pmt_container_output_20060102150405_*.log"))
```

Comment:
The time format is not really readable.
How about `pmt_container_output_2006-01-02_15-04-05_*.log `.

Reply:
Adjusted in 68f2ae8